kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/PodCrudTest.java

```diff
@@ -79,5 +86,298 @@ public void testCrud() {
+      @Override
+      public void eventReceived(Action action, Pod resource) {
+        switch (action) {
+          case DELETED:
+            deleteLatch.countDown();
+            break;
+          case MODIFIED:
+            editLatch.countDown();
+            break;
+          case ADDED:
+            addLatch.countDown();
+            break;
+          default:
+            throw new AssertionFailedError(action.toString().concat(" isn't recognised."));
+        }
+      }
+      @Override
+      public void onClose(KubernetesClientException cause) {
+        closeLatch.countDown();
+      }
+    });
+
+    pod1 = client.pods().inNamespace("ns1").withName(pod1.getMetadata().getName())
+      .patch(new PodBuilder().withNewMetadataLike(pod1.getMetadata()).endMetadata().build());
+
+    pod1.setSpec(new PodSpecBuilder().addNewContainer().withImage("nginx").withName("nginx").endContainer().build());
+
+    client.pods().inNamespace("ns1").withName(pod1.getMetadata().getName()).replace(pod1);
+
+    client.pods().inNamespace("ns1").withName(pod1.getMetadata().getName()).delete();
+
+    client.pods().inNamespace("ns1").create(new PodBuilder().withNewMetadata().withName("pod1").addToLabels("testKey", "testValue").endMetadata().build());
+
+    assertEquals(1, client.pods().inNamespace("ns1").list().getItems().size());
+    assertTrue(addLatch.await(1, TimeUnit.MINUTES));
+    assertTrue(editLatch.await(1, TimeUnit.MINUTES));
+    assertTrue(deleteLatch.await(1, TimeUnit.MINUTES));
+
+    watch.close();
+
+    assertTrue(closeLatch.await(1, TimeUnit.MINUTES));
+  }
+
+  @Test
+  public void testPodWatchOnNamespace() throws InterruptedException {
+    KubernetesClient client = server.getClient();
+    Pod pod1 = new PodBuilder().withNewMetadata().withName("pod1").addToLabels("testKey", "testValue").endMetadata().build();
+
+    final CountDownLatch deleteLatch = new CountDownLatch(3);
+    final CountDownLatch closeLatch = new CountDownLatch(3);
+    final CountDownLatch addLatch = new CountDownLatch(3);
+    final CountDownLatch editLatch = new CountDownLatch(2);
+
+    client.pods().inNamespace("ns1").create(pod1);
+    Watch watch = client.pods().inNamespace("ns1").watch(new Watcher<Pod>() {
+      @Override
+      public void eventReceived(Action action, Pod resource) {
+        switch (action) {
+          case DELETED:
+            deleteLatch.countDown();
+            break;
+          case ADDED:
+            addLatch.countDown();
+            break;
+          case MODIFIED:
+            editLatch.countDown();
+            break;
+          default:
+            throw new AssertionFailedError(action.toString());
+        }
+      }
+      @Override
+      public void onClose(KubernetesClientException cause) {
+        closeLatch.countDown();
+      }
+    });
+
+    Watch watch1 = client.pods().inNamespace("ns1").watch(new Watcher<Pod>() {
```

Comment:
I missed this during my first round of review. Why have you created 3 watches? I think one watch should be able to test all kind of events being emitted. Am I missing something here?